### PR TITLE
Add support for spherical meshes in ISOMIP+

### PR DESCRIPTION
**Pull request by xylar**
This merge builds on #469 and #470 to add support for meshes that are a patch of a sphere in ISOMIP+ test cases.

The idea is that these regional meshes on the sphere could be use to do idealized testing within E3SM.

### CODE REVIEW

**Comment by xylar** [comment]
# Tesitng

I have tested the spherical Ocean0 test but I haven't done thorough testing of any other tests (including the existing planar test cases) to make sure they work and are BFB.  That will follow once #469 and #470 have gone in.

**Comment by xylar** [comment]
To do:
* Add T62 mapping files
* use ELM's `coupler_bypass` approach to zero out atmosphere fluxes (grep).
* long name would have SROF
* `MPASO%CTEST` or similar in the long name

**Review by cbegeman — COMMENTED** [review]
@xylar Thanks for all your work on this! I ran the sphere/2km/z-star/Ocean0 test case. I did find this error at the streamfunction step

```
Traceback (most recent call last):
  File "/global/u1/c/cbegeman/compass-repos/add_isomip_plus_on_sphere/compass/run/serial.py", line 145, in run_tests
    _run_test(test_case)
  File "/global/u1/c/cbegeman/compass-repos/add_isomip_plus_on_sphere/compass/run/serial.py", line 394, in _run_test
    _run_step(test_case, step, test_case.new_step_log_file)
  File "/global/u1/c/cbegeman/compass-repos/add_isomip_plus_on_sphere/compass/run/serial.py", line 437, in _run_step
    step.run()
  File "/global/u1/c/cbegeman/compass-repos/add_isomip_plus_on_sphere/compass/ocean/tests/isomip_plus/streamfunction.py", line 74, in run
    _compute_overturning_streamfunction(dsMesh, ds, out_dir, dx=dx, dz=dz,
  File "/global/u1/c/cbegeman/compass-repos/add_isomip_plus_on_sphere/compass/ocean/tests/isomip_plus/streamfunction.py", line 141, in _compute_overturning_streamfunction
    _interpolate_horizontal_transport_zlevel(ds, z, zlevelTransportFileName,
  File "/global/u1/c/cbegeman/compass-repos/add_isomip_plus_on_sphere/compass/ocean/tests/isomip_plus/streamfunction.py", line 456, in _interpolate_horizontal_transport_zlevel
    assert(numpy.abs(dsOut.transportVertSumCheck).max().values < 1e-9)
AssertionError
  test execution:      ^[[91mERROR^[[0m
  test runtime:        ^[[94m20:19^[[0m
```

Is the streamfunction step supposed to be working for the sphere cases? It doesn't look like this error is related to the Cori filesystem issues, but I suppose it's possible. Also, this branch is from before your recent rebase.

**Comment by cbegeman** [comment]
@xylar I just tested the streamfunction step on your `fix_isomip_plus_land_ice_fraction` branch and at c139d0d9 and the error appears at both of them. So it doesn't appear to be introduced here.

File: compass/ocean/tests/isomip_plus/spherical_mesh.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        background_width = 100.
+
+        lat0 = self.config.getfloat('spherical_mesh', 'lat0')
+        fc = _make_feature(lat0)
+        fc.to_geojson('isomip_plus_high_res.geojson')
+
+        signed_distance = signed_distance_from_geojson(
+            fc, lon, lat, earth_radius, max_length=0.25)
+
+        trans_width = 1000e3
```
[the remark this answers is not in the document]
Comment (xylar) [reply compass/ocean/tests/isomip_plus/spherical_mesh.py:67]: This is a good suggestion.  I have added some comments on these two constants.

[the remark this answers is not in the document]
Comment (cbegeman) [reply compass/ocean/tests/isomip_plus/spherical_mesh.py:67]: Great, thanks!

**Comment by xylar** [comment]
@cbegeman, thanks for testing!  It sounds like you did a longer run than I did.  I ran for a month on Cori and the streamfunction step worked fine for me on that data.  How long were you running?  I assume longer because I think I recall one month taking something like 10 minutes but maybe I'm misremebering.

**Comment by cbegeman** [comment]
@xylar I was running the month-long simulation at 2km resolution. The interactive queue booted me off after some minutes of inactivity (key entry?) while running the simulation step on cori, so I just used the auto-generated job script and I think it took ~30 min.

**Comment by xylar** [comment]
@cbegeman, ah, I see. I think I'm getting confused by the key for generating green text that contains a 9.  The test case didn't actually take 91 minutes, then.

**Comment by cbegeman** [comment]
@xylar Oh, yeah, I think it says 20 min above.

**Comment by xylar** [comment]
@cbegeman, I incorrectly wrote Cori instead of Chrysalis above.  So a difference in my testing vs. yours might be the machine I was using.  I'm trying again on Chrysalis and will try Cori later on if Chrysalis doesn't show the issue for me in case it's machine-specific.

**Comment by cbegeman** [comment]
@xylar Thanks for doing that testing!

**Comment by xylar** [comment]
@cbegeman, I'm seeing the same on Chrysalis.  I'll look into it.  Thanks for your review so far.

**Comment by xylar** [comment]
@cbegeman, now that #548 is in, this should be ready for your review.

**Review by cbegeman — APPROVED** [review]
I ran the spherical Ocean0 test case successfully on chrys with intel, openmpi. The plots that I looked at (incl osf) were all good. Thanks @xylar!

**Comment by xylar** [comment]
Thanks @cbegeman!

**Pull request merged**